plugins/out_s3/s3_parquet.c

```diff
@@ -0,0 +1,500 @@
+                         batch->tag);
+            flb_free(parquet_buffer);
+            return -1;
+        }
+    } else if (parquet_size > MAX_FILE_SIZE_PUT_OBJECT) {
+        /* Use multipart upload for large files */
+        size_t offset = 0;
+        struct multipart_upload *local_upload = NULL;
+        int need_cleanup = FLB_FALSE;
+
+        flb_plg_debug(ctx->ins,
+                     "using multipart upload: %zu bytes",
+                     parquet_size);
+
+        /* Ensure we have a valid multipart upload structure */
+        if (!m_upload) {
+            /* Create a local multipart upload structure */
+            local_upload = create_upload(ctx, batch->tag, batch->tag_len,
+                                        batch->create_time);
+            if (!local_upload) {
+                flb_plg_error(ctx->ins,
+                             "failed to create multipart upload structure for batch '%s'",
+                             batch->tag);
+                flb_free(parquet_buffer);
+                return -1;
+            }
+            m_upload = local_upload;
+            need_cleanup = FLB_TRUE;
+        }
+
+        /* Ensure multipart upload is created */
+        if (m_upload->upload_state == MULTIPART_UPLOAD_STATE_NOT_CREATED) {
+            ret = create_multipart_upload(ctx, m_upload, NULL);
+            if (ret < 0) {
+                flb_plg_error(ctx->ins,
+                             "failed to create multipart upload for batch '%s'",
+                             batch->tag);
+                flb_free(parquet_buffer);
+                if (need_cleanup) {
+                    mk_list_del(&local_upload->_head);
+                    multipart_upload_destroy(local_upload);
+                }
+                return -1;
+            }
+            m_upload->upload_state = MULTIPART_UPLOAD_STATE_CREATED;
+        }
+
+        /* Upload each part */
+        while (offset < parquet_size) {
+            size_t chunk_size = (parquet_size - offset > ctx->upload_chunk_size) ?
+                               ctx->upload_chunk_size : (parquet_size - offset);
+
+            ret = upload_part(ctx, m_upload,
+                            parquet_buffer + offset, chunk_size, NULL);
+            if (ret < 0) {
+                flb_plg_error(ctx->ins,
+                             "failed to upload part for batch '%s'",
+                             batch->tag);
+                flb_free(parquet_buffer);
+                if (need_cleanup) {
+                    mk_list_del(&local_upload->_head);
+                    multipart_upload_destroy(local_upload);
+                }
+                return -1;
+            }
+
+            offset += chunk_size;
+            m_upload->part_number++;
+            /* Note: upload_part already updates m_upload->bytes, no need to add manually */
+        }
+
+        /* Mark for completion */
+        m_upload->upload_state = MULTIPART_UPLOAD_STATE_COMPLETE_IN_PROGRESS;
+
+    } else {
```

Comment:
We need to break a line like as:

```suggestion
    }
    else {
```